packages/common-stellar/src/project/models.ts

```diff
@@ -14,13 +14,100 @@ import {
   SubqlCustomDatasource,
   FileReference,
   CustomDataSourceAsset,
+  StellarBlockFilter,
+  StellarTransactionFilter,
+  StellarOperationFilter,
+  StellarEffectFilter,
+  SubqlBlockHandler,
+  SubqlTransactionHandler,
+  SubqlOperationHandler,
+  SubqlEffectHandler,
   SubqlEventHandler,
 } from '@subql/types-stellar';
 import {plainToClass, Transform, Type} from 'class-transformer';
 import {IsArray, IsEnum, IsInt, IsOptional, IsString, IsObject, ValidateNested} from 'class-validator';
+import {Horizon, ServerApi} from 'stellar-sdk';
 import {SubqlStellarProcessorOptions} from './types';
 
-export class EventFilter implements StellarEventFilter {
+export class BlockFilter implements StellarBlockFilter {
+  @IsOptional()
+  @IsInt()
+  modulo?: number;
+  @IsOptional()
+  @IsString()
+  timestamp?: string;
+}
```

Comment:
I guess we should move this to common (or a base of this) as its the same for all chains